core/base/src/TROOT.cxx

```diff
@@ -2158,46 +2157,37 @@ TClass *TROOT::LoadClass(const char *requestedname, Bool_t silent) const
 Int_t TROOT::LoadClass(const char * /*classname*/, const char *libname,
                        Bool_t check)
 {
-   Int_t err = -1;
-
-   char *path;
    TString lib = libname;
+   // Take care of user who didn't write the whole name
    if (!lib.BeginsWith("lib"))
       lib = "lib" + lib;
-   if ((path = gSystem->DynamicPathName(lib, kTRUE))) {
+
+   // Check if libname exists in path or not
+   if (char *path = gSystem->DynamicPathName(lib, kTRUE)) {
+      // If check == true, only check if it exists and if it's readable
       if (check)
-         err = 0;
+         return 0;
+
+      // If check == false, try to load the library
       else {
-         err = gSystem->Load(path, 0, kTRUE);
+        int err = gSystem->Load(path, 0, kTRUE);
+        // TSystem::Load returns 1 when the library was already loaded, return success in this case.
+        if (err == 1)
+          err = 0;
+        return err;
```

Comment:
What is path no longer delete? Isn't that a memory leak?

Reply:
path which is (char *) is a local variable on stack. It's a pointer to read-only data section in ROOT binary, and path is just a pointer to this. Am I missing something?

Reply:
Ah return value from DynamicPathName must be delete by user, I got your point